### PR TITLE
move start sqs queue in function invokation

### PR DESCRIPTION
**Pull request by EFF**

### CODE REVIEW

File: lib/LambdaHelper.js [hunk 1]
```diff
@@ -118,7 +123,7 @@ var _createProxy = function (executionStore, queueInitializedPromise, promise) {
 
         });
 
-        promise.catch(function () {
+        uploadPromise.catch(function () {
             throw "Could not upload function to S3";
```
Comment (slvnperron) [inline lib/LambdaHelper.js]: I know it's not part of the PR but this is crap. The throw is not part of the synchronous workflow

Comment (slvnperron) [inline lib/LambdaHelper.js]: i.e. we should simply console.log the error... eventually we should bubble up those errors as well

File: lib/SQSHelper.js [hunk 2]
```diff
@@ -60,9 +61,20 @@ module.exports = function (aws, executions){
...
+        _startListeningToSQS(_sqsQueueUrl, _sqsMessages);
+        _startPickingUpSQSMessages(_sqsQueueUrl, _sqsMessages);
+        this.initialized = true;
+        _initializedDeferred.resolve(_sqsQueueUrl);
+    }
+
     this.startQueue = function (startedCallback) {
         // TODO : move this in the function invocation ?
-        if (_initialized) throw "Already initialized";
+        // console.log(this.initialized);
```
Comment (slvnperron) [inline lib/SQSHelper.js]: remove these comments?

**Comment by slvnperron** [comment]
LGTM! Finally getting rid of this .start :+1:

File: lib/LambdaHelper.js (hunk 1 above)
Comment (EFF) [inline lib/LambdaHelper.js]: This is clearly crap!!

**Pull request merged**